pkg/resolver/search.go

```diff
@@ -265,321 +278,72 @@ func (s *SearchResult) resolveItems() ([]map[string]interface{}, error) {
-	case bool:
-		item = strconv.FormatBool(v)
-	case float64:
-		item = strconv.FormatInt(int64(v), 10)
-	default:
-		klog.Warningf("Error formatting property with type: %+v\n", reflect.TypeOf(v))
-	}
-	return item
-}
-
-func formatDataMap(data map[string]interface{}) map[string]interface{} {
-	item := make(map[string]interface{})
-	for key, value := range data {
-		switch v := value.(type) {
-		case string:
-			item[key] = v //strings.ToLower(v)
-		case bool:
-			item[key] = strconv.FormatBool(v)
-		case float64:
-			item[key] = strconv.FormatInt(int64(v), 10)
-		case map[string]interface{}:
-			item[key] = formatLabels(v)
-		case []interface{}:
-			item[key] = formatArray(v)
-		default:
-			klog.Warningf("Error formatting property with key: %+v  type: %+v\n", key, reflect.TypeOf(v))
-			continue
-		}
-	}
-	return item
-}
-
-// helper function to point values in string  array
-func pointerToStringArray(pointerArray []*string) []string {
-
-	values := make([]string, len(pointerArray))
-	for i, val := range pointerArray {
-		values[i] = *val
-	}
-	return values
-}
-
-func WhereClauseFilter(input *model.SearchInput) []exp.Expression {
 	var whereDs []exp.Expression
+	var dataTypeFromMap string
+	var err error
 
 	if input.Keywords != nil && len(input.Keywords) > 0 {
 		// Sample query: SELECT COUNT("uid") FROM "search"."resources", jsonb_each_text("data")
 		// WHERE (("value" LIKE '%dns%') AND ("data"->>'kind' ILIKE ANY ('{"pod","deployment"}')))
-		keywords := pointerToStringArray(input.Keywords)
+		keywords := PointerToStringArray(input.Keywords)
 		for _, key := range keywords {
 			key = "%" + key + "%"
 			whereDs = append(whereDs, goqu.L(`"value"`).ILike(key).Expression())
 		}
 	}
 	if input.Filters != nil {
+
 		for _, filter := range input.Filters {
 			if len(filter.Values) > 0 {
-				values := pointerToStringArray(filter.Values)
-
-				// If property is of array type like label, remove the equal sign in it and use colon
-				// - to be similar to how it is stored in the database
-				if _, ok := arrayProperties[filter.Property]; ok {
-					cleanedVal := make([]string, len(values))
-					for i, val := range values {
-						labels := strings.Split(val, "=")
-						if len(labels) == 2 {
-							cleanedVal[i] = fmt.Sprintf(`{"%s":"%s"}`, labels[0], labels[1])
-						} else if len(labels) == 1 {
-							//// If property is of array type, format it as an array for easy searching
-							cleanedVal[i] = labels[0]
-						} else {
-							klog.Error("Error while decoding label string")
-							cleanedVal[i] = val
+				values := PointerToStringArray(filter.Values)
+
+				if len(propTypeMap) > 0 {
+					if dataTypeInMap, ok := propTypeMap[filter.Property]; !ok { //check if value exists/if value doesn't exist
+						klog.Warningf("Property type for [%s] doesn't exist in cache. Refreshing property type cache",
+							filter.Property)
+						refresh := true
+						propTypeMapNew, err := GetPropertyTypeCache(ctx, refresh) //call database cache again
```

Comment:
No need to declare a new variable.
```suggestion
						propTypeMapNew, err := GetPropertyTypeCache(ctx, true) // call database cache again
```